cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -698,8 +705,6 @@ runTxBuild era (AnyConsensusModeParams cModeParams) networkId mScriptValidity
         & onNothing (left (ShelleyTxCmdEraConsensusModeMismatchTxBalance outputOptions
                             (AnyConsensusMode CardanoMode) (AnyCardanoEra era)))
 
-      SocketPath sockPath <- lift readEnvSocketPath & onLeft (left . ShelleyTxCmdSocketEnvError)
```

Comment:
Why are we not allowing specification of the socket path by environmental variable anymore?

Reply:
The above code is in the `runTxBuild` function.  This code previously read from the environment variable itself.  However, the caller of `runTxBuild` is `runTxBuildCmd` and that function already reads from the environment variable, so instead that function passes the socket file into `runTxBuild` so that we only read the environment variable once.  Doing so allows the caller to also decide not to read the environment variable, for example when the socket was provided explicitly on the command line.